File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/command/PreparePizzaCommand.java
package command;

class PreparePizzaCommand implements Command {
    private Receiver receiver;

    PreparePizzaCommand(Receiver receiver) {
        this.receiver = receiver;
    }

    @Override
    public void execute() {
        System.out.println("PreparePizzaCommand.execute(): Delegating to Receiver.preparePizza()");
        receiver.preparePizza();
    }
}

File: code/csharp/DesignPatterns/command/PreparePizzaCommand.cs
public class PreparePizzaCommand : ICommand
{
    private Receiver _receiver;

    public PreparePizzaCommand(Receiver receiver)
    {
        _receiver = receiver;
    }

    public void Execute()
    {
        System.Console.WriteLine("PreparePizzaCommand.Execute(): Delegating to Receiver.PreparePizza()");
        _receiver.PreparePizza();
    }
}
